sympy/simplify/simplify.py

```diff
@@ -610,6 +615,17 @@ def shorter(*choices):
     if measure(expr) > ratio*measure(original_expr):
         expr = original_expr
 
+    if rewrite:
+        rewritables = set()
+        for sub in preorder_traversal(original_expr):
+            rewritables.update([attr[17:] for attr in sub.func.__dict__.keys()
+                if attr[:17] == '_eval_rewrite_as_'])
+
+        rewritten = [simplify(original_expr.rewrite(func), ratio=ratio,
```

Comment:
- whether to rewrite original expression(`original_expr`) or simplified expression(`expr`)
- whether to check rewritten equations are identical to original expression or have duplicates: simplify() would give same results, does python or sympy caches the result?